bowtie2/AlignmentDocker.cwl

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env cwl-runner
+# This tool description was generated automatically by wdl2cwl ver. 0.2
+
+    cwlVersion: v1.0
+    baseCommand: []
+    requirements:         
+           -class: ShellCommandRequirement
+        
+        
+           -class: InlineJavascriptRequirement
+        
+        
+           -class: DockerRequirement
+             dockerPull: docker
+        
+    
+    class: CommandLineTool
+    inputs: 
+        leftFastq:
+            id: leftFastq
+            type: File
+        
+        rightFastq:
+            id: rightFastq
+            type: File
+        
+        ReferenceGenome:
+             id: ReferenceGenome
+             type: File
+        
+        sampleName:
+             id: sampleName
+             type: string
+        
+        index:
+             id: index
+             type: string
+        
+        docker:
+             id: docker
+             type: string
+        
+    
+    id: Alignment
+    outputs:
+        
+             id: rawSAM
+             type: File
+             outputBinding:
+                glob: $(inputs.sampleName).sam
+      
+    arguments: 
+             valueFrom: bowtie2-build $(inputs.ReferenceGenome.path) $(inputs.index)    bowtie2 -q -x $(inputs.index) -1 $(inputs.leftFastq.path) -2 $(inputs.rightFastq.path) -S $(inputs.sampleName).sam
+             shellQuote: false
```

Comment:
```suggestion
    arguments: 
             valueFrom: bowtie2-build $(inputs.ReferenceGenome.path) $(inputs.index)  |  bowtie2 -q -x $(inputs.index) -1 $(inputs.leftFastq.path) -2 $(inputs.rightFastq.path) -S $(inputs.sampleName).sam
             shellQuote: false
```

Reply:
Piped bowtie2-bild and bowtie2.

Reply:
Please validate the change.
There is error.
```
(env) ngsap2@ubuntu:~$ cwl-runner alignmentDocker.cwl 
INFO /home/ngsap2/.local/bin/cwl-runner 3.0.20200324120055
INFO Resolved 'alignmentDocker.cwl' to 'file:///home/ngsap2/alignmentDocker.cwl'
ERROR Tool definition failed validation:
alignmentDocker.cwl:14:24: mapping values are not allowed here
```

Reply:
I have run cwl-format and cwltool --validate for AlignmentDocker.cwl. But cwltool run fails. Error is below.
````
ngsap2@ubuntu:~$ sudo docker run -v /var/run/docker.sock:/var/run/docker.sock -v "$PWD":"$PWD" -w="$PWD" commonworkflowlanguage/cwltool /home/ngsap2/samtoolsDocker/AlignmentDockerNew.cwl --leftFastq /home/ngsap2/Downloads/testdata16march/ebola_mut_reads1.fq --rightFastq /home/ngsap2/Downloads/testdata16march/ebola_mut_reads2.fq --ReferenceGenome /home/ngsap2/Downloads/testdata16march/ebola_ref.fasta --sampleName ebola --index eb --docker comics/bowtie2
INFO /usr/local/bin/cwltool 2.0.20200107113851
INFO Resolved '/home/ngsap2/samtoolsDocker/AlignmentDockerNew.cwl' to 'file:///home/ngsap2/samtoolsDocker/AlignmentDockerNew.cwl'
INFO ['docker', 'pull', 'docker']
Using default tag: latest
latest: Pulling from library/docker
cbdbe7a5bc2a: Pulling fs layer
6f4c731d6eb3: Pulling fs layer
5d79d4ac3cd0: Pulling fs layer
65de10866d6a: Pulling fs layer
ffd69ed7f17f: Pulling fs layer
07ba9e5a2fa6: Pulling fs layer
75478c283dd2: Pulling fs layer
ffd69ed7f17f: Waiting
07ba9e5a2fa6: Waiting
75478c283dd2: Waiting
65de10866d6a: Waiting
5d79d4ac3cd0: Verifying Checksum
5d79d4ac3cd0: Download complete
cbdbe7a5bc2a: Verifying Checksum
cbdbe7a5bc2a: Download complete
6f4c731d6eb3: Verifying Checksum
6f4c731d6eb3: Download complete
cbdbe7a5bc2a: Pull complete
ffd69ed7f17f: Download complete
07ba9e5a2fa6: Verifying Checksum
07ba9e5a2fa6: Download complete
6f4c731d6eb3: Pull complete
5d79d4ac3cd0: Pull complete
75478c283dd2: Verifying Checksum
75478c283dd2: Download complete
65de10866d6a: Verifying Checksum
65de10866d6a: Download complete
65de10866d6a: Pull complete
ffd69ed7f17f: Pull complete
07ba9e5a2fa6: Pull complete
75478c283dd2: Pull complete
Digest: sha256:8c119115801ae96e1f55b2ccb69cc84537c79eea057a00f363df0196f3d09fd4
Status: Downloaded newer image for docker:latest
INFO [job AlignmentDocker] /tmp/jx5vfv7x$ docker \
    run \
    -i \
    --volume=/tmp/jx5vfv7x:/BScJye:rw \
    --volume=/tmp/e3m0q78a:/tmp:rw \
    --volume=/home/ngsap2/Downloads/testdata16march/ebola_mut_reads1.fq:/var/lib/cwl/stgfeed78c8-d5f0-4802-9b1d-d8e9ab026a04/ebola_mut_reads1.fq:ro \
    --volume=/home/ngsap2/Downloads/testdata16march/ebola_mut_reads2.fq:/var/lib/cwl/stg425eea77-c9d2-44d6-b616-53e511a4b32d/ebola_mut_reads2.fq:ro \
    --volume=/home/ngsap2/Downloads/testdata16march/ebola_ref.fasta:/var/lib/cwl/stg5c94f135-a5af-4a66-8fe8-fb0d1541a958/ebola_ref.fasta:ro \
    --workdir=/BScJye \
    --read-only=true \
    --user=0:0 \
    --rm \
    --env=TMPDIR=/tmp \
    --env=HOME=/BScJye \
    --cidfile=/tmp/b66i6_y0/20200517054753-879463.cid \
    docker \
    /bin/sh \
    -c \
    bowtie2-build /var/lib/cwl/stg5c94f135-a5af-4a66-8fe8-fb0d1541a958/ebola_ref.fasta eb | bowtie2 -q -x eb -1 /var/lib/cwl/stgfeed78c8-d5f0-4802-9b1d-d8e9ab026a04/ebola_mut_reads1.fq -2 /var/lib/cwl/stg425eea77-c9d2-44d6-b616-53e511a4b32d/ebola_mut_reads2.fq -S ebola.sam
/bin/sh: bowtie2-build: not found
/bin/sh: bowtie2: not found
INFO [job AlignmentDocker] Max memory used: 0MiB
ERROR [job AlignmentDocker] Job error:
("Error collecting output for parameter 'rawSAM':\nsamtoolsDocker/AlignmentDockerNew.cwl:31:5: Did not find output file with glob pattern: '['ebola.sam']'", {})
WARNING [job AlignmentDocker] completed permanentFail
{}
WARNING Final process status is permanentFail
````